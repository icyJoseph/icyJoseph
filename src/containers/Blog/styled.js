import styled from "styled-components";

export const BlogWrap = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: center;

  @media (max-width: 599px) {
    justify-content: flex-start;
    margin-top: 1em;
    margin-bottom: 4em;
  }
`;

BlogWrap.Info = styled.p`
  width: 40%;
  text-align: center;
`;

BlogWrap.Icon = styled.div`
  font-size: 2em;
`;

BlogWrap.Content = styled.div`
  margin: 0 auto;
  display: flex;
  flex-direction: column;

  > * {
    margin: 0.5em;
    justify-self: center;
    align-self: center;
  }
`;

BlogWrap.Link = styled.a`
  background: ${({ theme }) => theme.background};
  color: white;
  margin-top: 2em;
  padding: 0.3em 0.5em;
  position: relative;
  text-decoration: none;
  border-radius: 0.3em;
  box-shadow: 0 1px 7px rgba(0, 0, 0, 0.22), 0 1px 6px rgba(0, 0, 0, 0.34);

  &:hover {
    cursor: pointer;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.35), 0 7px 15px rgba(0, 0, 0, 0.32),
      0 -3px 15px rgba(0, 0, 0, 0.32);
  }
`;
