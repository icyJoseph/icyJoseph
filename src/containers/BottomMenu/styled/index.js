import styled from "styled-components";
import Menu from "semantic-ui-react/dist/commonjs/collections/Menu";

export const MenuWrapper = styled(Menu)`
  width: 100%;
  border-radius: 0;
  position: fixed;
  bottom: 0;
`;
