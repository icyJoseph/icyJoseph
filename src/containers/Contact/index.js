import React from "react";
import PropTypes from "prop-types";
import { Header, Modal, Button } from "semantic-ui-react";
import { ModalWrapper } from "./styled";

export const Contact = ({ open, handleClose }) => (
  <ModalWrapper
    dimmer={true}
    open={open}
    onClose={handleClose}
    closeOnDocumentClick
  >
    <Modal.Header>Contact</Modal.Header>
    <Modal.Content>
      <Modal.Description>
        <Header>Email</Header>
        <p>You can reach me via email to:</p>
        <p>contact@icjoseph.com</p>
      </Modal.Description>
    </Modal.Content>
    <Modal.Actions>
      <Button color="black" onClick={handleClose}>
        Close
      </Button>
    </Modal.Actions>
  </ModalWrapper>
);

export default Contact;

Contact.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func
};
