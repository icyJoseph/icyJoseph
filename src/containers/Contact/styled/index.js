import styled from "styled-components";
import { Modal } from "semantic-ui-react";

export const ModalWrapper = styled(Modal)`
  @media (max-width: 1023px) {
    width: 100%;
    margintop: 33.33%;
  }
`;
