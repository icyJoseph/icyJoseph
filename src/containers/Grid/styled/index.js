import styled from "styled-components";
import { Sidebar } from "semantic-ui-react";

export const Pushable = styled(Sidebar.Pushable)`
  height: 100vh;
  border: none;
  box-shadow: none;
  margin-top: 0px;
  border-radius: 0;
  background-image: ${props => `url(${props.background})`};
  background-size: 100% 100%;
`;
