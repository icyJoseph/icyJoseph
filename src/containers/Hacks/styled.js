import styled from "styled-components";

export const RepoWrap = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 1em 0.5em;
  padding: 0.25em 0.5em;
`;

export const FilterWrap = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
`;

export const HacksWrap = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 4em;
`;
