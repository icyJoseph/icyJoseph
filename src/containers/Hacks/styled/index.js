import styled from "styled-components";
import { Sidebar } from "semantic-ui-react";

export const Pushable = styled(Sidebar.Pushable)`
  border: none;
  box-shadow: none;
  margin-top: 0px;
  border-radius: 0;
  height: 100%;
  position: fixed;
  width: 100%;
`;
