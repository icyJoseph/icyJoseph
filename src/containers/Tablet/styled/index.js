import styled from "styled-components";

export const TabletWrapper = styled.div`
  background: dodgerblue;
  height: 100%;
  padding: 10%;
  margin-bottom: 15px;
`;
