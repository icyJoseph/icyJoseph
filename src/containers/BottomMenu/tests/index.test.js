import React from "react";
import { shallow } from "enzyme";
import { Menu, Button, Modal } from "semantic-ui-react";
import { BottomMenu } from "../";
import { Contact } from "../../Contact";
import { ModalWrapper } from "../../Contact/styled";

describe("BottomMenu", () => {
  const spyItemClick = jest.spyOn(BottomMenu.prototype, "handleItemClick");
  const spyClickContact = jest.spyOn(
    BottomMenu.prototype,
    "handleClickContact"
  );
  const spyCloseContact = jest.spyOn(
    BottomMenu.prototype,
    "handleCloseContact"
  );

  const wrapper = shallow(<BottomMenu />);

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has no active item in the beginning", () => {
    expect(wrapper.state("activeItem")).toBe(undefined);
    expect(wrapper.state("contact")).toEqual(false);
  });

  it("sets active item", () => {
    wrapper
      .find(Menu.Item)
      .at(0)
      .simulate(
        "click",
        { event: { target: { value: "" } } },
        { name: "LinkedIn" }
      );
    expect(wrapper.state("activeItem")).toEqual("LinkedIn");
    expect(spyItemClick).toHaveBeenCalled();
  });

  it("sets active contact", () => {
    wrapper
      .find(Menu.Item)
      .at(2)
      .simulate(
        "click",
        { event: { target: { value: "" } } },
        { name: "Contact" }
      );
    expect(wrapper.state("activeItem")).toEqual("Contact");
    expect(wrapper.state("contact")).toEqual(true);
    expect(spyClickContact).toHaveBeenCalled();
  });

  it("closes contact", () => {
    expect(wrapper.find(Contact).prop("open")).toEqual(
      wrapper.state("contact")
    );

    wrapper
      .find(Contact)
      .dive()
      .find(Button)
      .simulate("click");

    expect(wrapper.state("contact")).toEqual(false);
    expect(spyCloseContact).toHaveBeenCalled();
  });
});
