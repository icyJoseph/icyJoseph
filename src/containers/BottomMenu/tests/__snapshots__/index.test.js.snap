// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomMenu renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BottomMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <styled.div>
          <Button
            as="button"
            circular={true}
            color="black"
            href="https://github.com/icyJoseph"
            icon={
              <Icon
                as="i"
                name="github alternate"
                size="large"
              />
            }
            rel="noopener noreferrer"
            role="button"
            size="huge"
            style={
              Object {
                "margin": "5px",
              }
            }
            target="_blank"
          />
          <Button
            as="button"
            circular={true}
            color="black"
            href="https://medium.com/@icjoseph"
            icon={
              <Icon
                as="i"
                name="medium m"
                size="large"
              />
            }
            rel="noopener noreferrer"
            role="button"
            size="huge"
            style={
              Object {
                "margin": "5px",
              }
            }
            target="_blank"
          />
          <Button
            as="button"
            circular={true}
            color="black"
            icon={
              <Icon
                as="i"
                name="mail"
                size="large"
              />
            }
            onClick={[Function]}
            role="button"
            size="huge"
            style={
              Object {
                "margin": "5px",
              }
            }
          />
        </styled.div>,
        <Contact
          handleClose={[Function]}
          open={false}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              as="button"
              circular={true}
              color="black"
              href="https://github.com/icyJoseph"
              icon={
                <Icon
                  as="i"
                  name="github alternate"
                  size="large"
                />
              }
              rel="noopener noreferrer"
              role="button"
              size="huge"
              style={
                Object {
                  "margin": "5px",
                }
              }
              target="_blank"
            />,
            <Button
              as="button"
              circular={true}
              color="black"
              href="https://medium.com/@icjoseph"
              icon={
                <Icon
                  as="i"
                  name="medium m"
                  size="large"
                />
              }
              rel="noopener noreferrer"
              role="button"
              size="huge"
              style={
                Object {
                  "margin": "5px",
                }
              }
              target="_blank"
            />,
            <Button
              as="button"
              circular={true}
              color="black"
              icon={
                <Icon
                  as="i"
                  name="mail"
                  size="large"
                />
              }
              onClick={[Function]}
              role="button"
              size="huge"
              style={
                Object {
                  "margin": "5px",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "circular": true,
              "color": "black",
              "href": "https://github.com/icyJoseph",
              "icon": <Icon
                as="i"
                name="github alternate"
                size="large"
              />,
              "rel": "noopener noreferrer",
              "role": "button",
              "size": "huge",
              "style": Object {
                "margin": "5px",
              },
              "target": "_blank",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "circular": true,
              "color": "black",
              "href": "https://medium.com/@icjoseph",
              "icon": <Icon
                as="i"
                name="medium m"
                size="large"
              />,
              "rel": "noopener noreferrer",
              "role": "button",
              "size": "huge",
              "style": Object {
                "margin": "5px",
              },
              "target": "_blank",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "circular": true,
              "color": "black",
              "icon": <Icon
                as="i"
                name="mail"
                size="large"
              />,
              "onClick": [Function],
              "role": "button",
              "size": "huge",
              "style": Object {
                "margin": "5px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <styled.div>
            <Button
              as="button"
              circular={true}
              color="black"
              href="https://github.com/icyJoseph"
              icon={
                <Icon
                  as="i"
                  name="github alternate"
                  size="large"
                />
              }
              rel="noopener noreferrer"
              role="button"
              size="huge"
              style={
                Object {
                  "margin": "5px",
                }
              }
              target="_blank"
            />
            <Button
              as="button"
              circular={true}
              color="black"
              href="https://medium.com/@icjoseph"
              icon={
                <Icon
                  as="i"
                  name="medium m"
                  size="large"
                />
              }
              rel="noopener noreferrer"
              role="button"
              size="huge"
              style={
                Object {
                  "margin": "5px",
                }
              }
              target="_blank"
            />
            <Button
              as="button"
              circular={true}
              color="black"
              icon={
                <Icon
                  as="i"
                  name="mail"
                  size="large"
                />
              }
              onClick={[Function]}
              role="button"
              size="huge"
              style={
                Object {
                  "margin": "5px",
                }
              }
            />
          </styled.div>,
          <Contact
            handleClose={[Function]}
            open={false}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                as="button"
                circular={true}
                color="black"
                href="https://github.com/icyJoseph"
                icon={
                  <Icon
                    as="i"
                    name="github alternate"
                    size="large"
                  />
                }
                rel="noopener noreferrer"
                role="button"
                size="huge"
                style={
                  Object {
                    "margin": "5px",
                  }
                }
                target="_blank"
              />,
              <Button
                as="button"
                circular={true}
                color="black"
                href="https://medium.com/@icjoseph"
                icon={
                  <Icon
                    as="i"
                    name="medium m"
                    size="large"
                  />
                }
                rel="noopener noreferrer"
                role="button"
                size="huge"
                style={
                  Object {
                    "margin": "5px",
                  }
                }
                target="_blank"
              />,
              <Button
                as="button"
                circular={true}
                color="black"
                icon={
                  <Icon
                    as="i"
                    name="mail"
                    size="large"
                  />
                }
                onClick={[Function]}
                role="button"
                size="huge"
                style={
                  Object {
                    "margin": "5px",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "circular": true,
                "color": "black",
                "href": "https://github.com/icyJoseph",
                "icon": <Icon
                  as="i"
                  name="github alternate"
                  size="large"
                />,
                "rel": "noopener noreferrer",
                "role": "button",
                "size": "huge",
                "style": Object {
                  "margin": "5px",
                },
                "target": "_blank",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "circular": true,
                "color": "black",
                "href": "https://medium.com/@icjoseph",
                "icon": <Icon
                  as="i"
                  name="medium m"
                  size="large"
                />,
                "rel": "noopener noreferrer",
                "role": "button",
                "size": "huge",
                "style": Object {
                  "margin": "5px",
                },
                "target": "_blank",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "circular": true,
                "color": "black",
                "icon": <Icon
                  as="i"
                  name="mail"
                  size="large"
                />,
                "onClick": [Function],
                "role": "button",
                "size": "huge",
                "style": Object {
                  "margin": "5px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
